### PR TITLE
Let's Encrypt certificates

### LINKED ISSUE #117: Add support for LetsEncrypt endpoints
See https://github.com/dnsimple/dnsimple-developer/pull/152

### PR DESCRIPTION
**Pull request by weppos**
Implements https://github.com/dnsimple/dnsimple-developer/pull/152

---

Closes #117

### CODE REVIEW

File: lib/dnsimple/certificates.ex [hunk 1]
```diff
@@ -33,18 +34,18 @@ defmodule Dnsimple.Certificates do
...
+  - https://developer.dnsimple.com/v2/domains/certificates/#getCertificate
 
   ## Examples
 
       client = %Dnsimple.Client{access_token: "a1b2c3d4"}
       {:ok, response} = Dnsimple.Certificates.get_certificate(client, account_id = "1010", domain_id = "example.com")
 
   """
-  @spec get_certificate(Client.t, String.t | integer, String.t | integer, String.t | integer, Keyword.t) :: {:ok|:errorm, Response.t}
+  @spec get_certificate(Client.t, String.t | integer, String.t | integer, integer, Keyword.t) :: {:ok|:errorm, Response.t}
```
Comment (jacegu) [inline lib/dnsimple/certificates.ex]: Typo in the result type specification ⚠️ `errorm` (was there already, but I just noticed it during this review).

File: lib/dnsimple/certificates.ex [hunk 2]
```diff
@@ -75,23 +76,91 @@ defmodule Dnsimple.Certificates do
...
+
+  @doc """
+  Purchase a Let's Encrypt certificate.
+
+  This method creates a new certificate order. The certificate ID should be used to
+  request the issuance of the certificate using `#purchase_letsencrypt_certificate`.
+
+  See:
+  - https://developer.dnsimple.com/v2/domains/certificates/#purchaseLetsencryptCertificate
+  """
```
Comment (jacegu) [inline lib/dnsimple/certificates.ex]: We have examples for all other endpoints (unless we missed any). Do you mind providing some in these newly added functions?

File: lib/dnsimple/certificates.ex [hunk 3]
```diff
@@ -75,23 +76,91 @@ defmodule Dnsimple.Certificates do
...
+    url = Client.versioned("/#{account_id}/domains/#{domain_id}/certificates/letsencrypt/#{certificate_id}/issue")
+
+    Client.post(client, url, Client.empty_body(), options)
+    |> Response.parse(%{"data" => %Certificate{}})
+  end
+
+
+  @doc """
+  Purchase a Let's Encrypt certificate renewal.
+
```
Comment (jacegu) [inline lib/dnsimple/certificates.ex]: I'd add a similar note to the one you added in purchasing in here:

>   This method creates a new certificate order. The certificate ID should be used to request the issuance of the certificate using `#purchase_letsencrypt_certificate`

File: lib/dnsimple/certificates.ex [hunk 4]
```diff
@@ -75,23 +76,91 @@ defmodule Dnsimple.Certificates do
...
+  See:
+  - https://developer.dnsimple.com/v2/domains/certificates/#purchaseRenewalLetsencryptCertificate
+  """
+  @spec purchase_letsencrypt_certificate_renewal(Client.t, String.t | integer, String.t | integer, integer, Keyword.t, Keyword.t) :: {:ok|:error, Response.t}
+  def purchase_letsencrypt_certificate_renewal(client, account_id, domain_id, certificate_id, attributes, options \\ []) do
```
Comment (jacegu) [inline lib/dnsimple/certificates.ex]: Requiring attributes to be passed in makes me think there is a mandatory one. However, [looking at the docs](https://github.com/dnsimple/dnsimple-developer/pull/152/files#diff-cc600a305cca05054037bf601396251eR300), the only attribute that applies here is `auto_renew` and it's not mandatory. Are the docs or the code correct?

Comment (jodosha) [reply lib/dnsimple/certificates.ex]: @jacegu `auto_renew` isn't mandatory. Docs are correct. If omitted, the server applies `false` to that field.

Comment (jacegu) [reply lib/dnsimple/certificates.ex]: Then I would not require customers to pass in an attribute map and I'd default `attributes` to `%{}`. We are doing this in other endpoints when the situation is the same: POST/PUT with only optional attributes.

File: lib/dnsimple/certificate.ex [hunk 5]
```diff
@@ -16,23 +16,27 @@ defmodule Dnsimple.Certificate do
   @type t :: %__MODULE__{
     id: integer,
     domain_id: integer,
-    name: String.t,
```
Comment (OleMchls) [inline lib/dnsimple/certificate.ex]: This causes a breaking change we will need to communicate.

File: lib/dnsimple/certificates.ex (hunk 2 above)
Comment (weppos) [reply lib/dnsimple/certificates.ex]: ✔️

File: lib/dnsimple/certificates.ex (hunk 3 above)
Comment (weppos) [reply lib/dnsimple/certificates.ex]: ✔️

File: lib/dnsimple/certificates.ex (hunk 4 above)
Comment (weppos) [reply lib/dnsimple/certificates.ex]: ✔️

File: lib/dnsimple/certificate.ex (hunk 5 above)
Comment (weppos) [reply lib/dnsimple/certificate.ex]: Yes, I'll add a note in the changelog.

**Comment by weppos** [comment]
I made all the requested changes. Please confirm the PR.

**Pull request merged**